cura/CuraApplication.py

```diff
@@ -1940,6 +1947,11 @@ def shouldShowWelcomeDialog(self) -> bool:
         # Only show the complete flow if there is no printer yet.
         return self._machine_manager.activeMachine is None
 
+    @pyqtSlot(result = bool)
+    def shouldShowAddPrintersUncancellableDialog(self) -> bool:
```

Comment:
I don't really see a reason for this to be in the Cura application (or in python for that matter). Both the required properties are already exposed in QML, so why not make it part of that?